### PR TITLE
Refactor `StreamWrapper` and raise psalm level to 4

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | partly #204

### CODE REVIEW

File: src/Collector/Stream/FilesystemStreamProxy.php [hunk 1]
```diff
@@ -17,7 +17,7 @@ class FilesystemStreamProxy implements StreamWrapperInterface
      * @var resource|null
      */
     public $context;
-    public StreamWrapperInterface $decorated;
+    public StreamWrapper $decorated;
```
Comment (xepozz) [inline src/Collector/Stream/FilesystemStreamProxy.php:20]: For what reason?

File: src/Collector/Stream/HttpStreamProxy.php [hunk 2]
```diff
@@ -20,7 +20,7 @@ class HttpStreamProxy implements StreamWrapperInterface
      * @var resource|null
      */
     public $context;
-    public StreamWrapperInterface $decorated;
+    public StreamWrapper $decorated;
```
Comment (xepozz) [inline src/Collector/Stream/HttpStreamProxy.php:23]: Same above

**Review by xepozz — CHANGES_REQUESTED** [review]
Did you test the changes in src/Helper/StreamWrapper/StreamWrapper.php?
It worked in many cases I've tested and I'm not sure it's good way to change the logic without integrational tests. 
PHP documentation is poor in streams context and parts of the source were taken from many other open source projects that works well. So it's better keep it as is.

File: src/Collector/Stream/FilesystemStreamProxy.php (hunk 1 above)
Comment (vjik) [reply src/Collector/Stream/FilesystemStreamProxy.php:20]: I assumed that `$decorated` setted in constructor only (used `StreamWrapper`).

**Comment by vjik** [comment]
> It worked in many cases I've tested and I'm not sure it's good way to change the logic without integrational tests.
> PHP documentation is poor in streams context and parts of the source were taken from many other open source projects that works well. So it's better keep it as is.

Agree, need test. Suggest do it in separate PR. And now this package in development stage and if there are errors, then we will have the opportunity to fix them.

**Comment by xepozz** [comment]
So would you like to separate the changes from `src/Helper/StreamWrapper/StreamWrapper.php` into a separated PR?

**Pull request closed without merging**